### PR TITLE
Implementation/45520 show folder without create permission as inactive

### PR DESCRIPTION
**Pull request by apfohl**

### CODE REVIEW

File: frontend/src/app/core/state/storage-files/storage-files.service.ts [hunk 1]
```diff
@@ -60,6 +60,7 @@ export class StorageFilesResourceService extends ResourceCollectionService<IStor
           insertCollectionIntoState(this.store, collection, link.href);
         }),
         map((collection) => collection._embedded.elements),
+        take(1),
```
Comment (Kharonus) [inline frontend/src/app/core/state/storage-files/storage-files.service.ts:63]: Do we really need to move it here?

File: frontend/src/app/shared/components/storages/file-picker-base-modal.component.ts/file-picker-base-modal.component.ts [hunk 2]
```diff
@@ -56,15 +56,17 @@ import {
...
-      navigate: () => this.changeLevel(null, this.breadcrumbs.crumbs.slice(0, 1)),
-    }],
-  );
+  protected currentDirectory:IStorageFile;
+
+  protected get storageLink():IHalResourceLink {
+    return this.locals.storageLink as IHalResourceLink;
+  }
+
+  public breadcrumbs:BreadcrumbsContent = new BreadcrumbsContent([]);
```
Comment (Kharonus) [inline frontend/src/app/shared/components/storages/file-picker-base-modal.component.ts/file-picker-base-modal.component.ts]: This does not fit, IMHO. the breadcrumbs need some kind of first value, without it, the GUI will flicker. Right now you write the breadcrumbs only after the files are fetched (loading skeleton), which will take some time as an outbound request.

I suggest seeding the crumbs with a dummy crumb, which has only an icon and a text.

File: frontend/src/app/shared/components/storages/storage-file-list-item/storage-file-list-item.ts [hunk 3]
```diff
@@ -60,16 +60,24 @@ export class StorageFileListItem {
...
+  get isReadable():boolean {
+    return this.storageFile.permissions.some((permission) => permission === 'readable');
   }
 
   constructor(
     private readonly timezoneService:TimezoneService,
     private readonly storageFile:IStorageFile,
     public readonly disabled:boolean,
     public readonly isFirst:boolean,
+    public readonly enterDirectory:() => void,
```
Comment (Kharonus) [inline frontend/src/app/shared/components/storages/storage-file-list-item/storage-file-list-item.ts:79]: what is the reason to make this callback mandatory?

File: frontend/src/app/core/state/storage-files/storage-files.service.ts (hunk 1 above)
Comment (Kharonus) [reply frontend/src/app/core/state/storage-files/storage-files.service.ts:63]: I think it is fine, it basically removed 2 instances in the caller, but replaced it by a single one here.

**Review by Kharonus — CHANGES_REQUESTED** [review]

File: frontend/src/app/core/state/storage-files/storage-files.service.ts (hunk 1 above)
Comment (apfohl) [reply frontend/src/app/core/state/storage-files/storage-files.service.ts:63]: It is not needed per se. It's more a case of, how we understand an handle Observables. This one behaves now like a Promise. But handling must be generalized.

File: frontend/src/app/shared/components/storages/storage-file-list-item/storage-file-list-item.ts (hunk 3 above)
Comment (apfohl) [reply frontend/src/app/shared/components/storages/storage-file-list-item/storage-file-list-item.ts:79]: In every code path it is set. It's no longer `nil`. The default is now `() => {}`. Therefore it does not need to be optional.

File: frontend/src/app/shared/components/storages/file-picker-base-modal.component.ts/file-picker-base-modal.component.ts (hunk 2 above)
Comment (apfohl) [reply frontend/src/app/shared/components/storages/file-picker-base-modal.component.ts/file-picker-base-modal.component.ts]: I agree. I renders so fast on my machine, that I overlooked it.

File: frontend/src/app/shared/components/storages/storage-file-list-item/storage-file-list-item.ts (hunk 3 above)
Comment (Kharonus) [reply frontend/src/app/shared/components/storages/storage-file-list-item/storage-file-list-item.ts:79]: yeah, question would be, why it is not `undefined` for a non directory item. But actually, I doesn't care too much :D

File: frontend/src/app/shared/components/storages/file-picker-base-modal.component.ts/file-picker-base-modal.component.ts (hunk 2 above)
Comment (apfohl) [reply frontend/src/app/shared/components/storages/file-picker-base-modal.component.ts/file-picker-base-modal.component.ts]: Fixed.

**Review by Kharonus — APPROVED** [review]
LGTM

**Pull request merged**